src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(health = 100, name, hidden = false) {
+    this.health = health;
+    this.name = name;
+    this.hidden = hidden;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(100, name, hidden);
+  }
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(100, name, hidden);
+  }
+
+  bite(prey) {
+    if (prey instanceof Herbivore) {
+      if (!prey.hidden) {
+        prey.health -= 50;
+      }
+
+      if (prey.health <= 0) {
+        const index = Animal.alive.indexOf(prey);
+
+        if (index !== -1) {
+          Animal.alive.splice(index, 1);
```

Comment:
This verification and removal of the Animal (prey) should be added to the animal class as a better practice: think about it, if you needed to make another animal die / check it, you would need to reuse this whole code